### ISSUE TITLE
mpc-qt backup

### ISSUE THREAD

**Issue by wm4** [ahjolinna/mpc-qt#1]
Unfortunately, @cmdkotori removed himself and his project from github. (I hope he's OK.) Up-to-date forks of mpc-qt are unfortunately rare. I think my copy might be relatively recent: https://github.com/wm4/mpc-qt
This repo is possibly newer (some minor code changes and a change to the README, probably 2 commits), but lacks a history.
Well, I'll leave it at that.

**Comment by ansraliant** [comment]
Hi @wm4, thanks for reaching out. 
Your repo seems to have almost all the history, perhaps we can get merge it to get the history while keeping the latest changes. What do you think @ahjolinna ?
As far as I know, the repo from @ahjolinna was 1 or 2 days before @cmdkotori removed the project from github.
Thanks

**Comment by ahjolinna** [comment]
Mine should be the latest copy before removed his account/project, as I updated my [OBS/openSUSE repo](https://build.opensuse.org/package/show/home:ahjolinna/mpc-qt) just less than 1 day before.

When it comes to merging I dont care one way or another I just released this for everyone who liked mpc-qt.
I mostly care who will take over the project. At least some basic maintenance to support latest Qt and mpv changes.

btw. day before @cmdkotori left had made a bug report where he said that he was burnout, also he had some other personal issues. Unfortunately he also removed his email...so yeah

**Comment by luigino** [comment]
I checked all the existing mpc-qt forks for the most recent and diffed it with the backup ahjolinna uploaded.

The result is on https://gitlab.com/lbaldoni/mpc-qt .

P.S: I tried contacting CmdKotori but he deleted his mailbox as well.

**Comment by wm4** [comment]
That shows way more changes than my attempts showed. Will have to recheck.

**Comment by wm4** [comment]
OK, looked at it. My repo has 14 commits more then yours. Note that I hadn't update my github fork until I 2 days ago when I noticed that cmdkotori removed himself and mpc-qt from the internet.

There are some minor changes between ahjolinna's repo and mine. But the small volume of changes, uncertain copyright makes my history is the best to start from. (It's almost 100% certain cmdrkotori made these changes, but it's not 100% certain.)

**Comment by flopcat** [comment]
https://github.com/flopcat/mpc-qt has everything i think

**Comment by wm4** [comment]
@flopcat I think your repo wins. @ahjolinna has some missing .gitignore (possibly from a copy command that skipped "hidden" files), and an edit to README.md that adds a screenshot.
I now updated my repo to your state.
I haven't noticed your fork when I checked all forks, did you push to it later?

**Comment by flopcat** [comment]
yes i pushed it when i saw this conversation

**Comment by luigino** [comment]
Should whoever wants to keep maintaining this perhaps create an "official" organisation?
Maybe somewhere else than github too.

**Comment by wm4** [comment]
Well that's the problem. I haven't seen or heard of anyone yet who want to step up as a maintainer.
But it's true that there should be some sort of main repository to avoid fragmentation.

**Comment by luigino** [comment]
Meanwhile I created this: https://gitlab.com/mpc-qt/mpc-qt.
I can follow minor things but I'm not able to maintain the main codebase, so if any of you gentlemen want to join, plese do.

**Comment by MeAKAErik** [comment]
Posting just to say I hope Kotori is ok and wish them the best if they are reading this.  I remember seeing @ahjolinna and Kotori talking about some sort of medical problem in one of the project's previous issues so I hope it is not related to that.

**Comment by maor26** [comment]
Any news about the project? was looking for a new player ans saw this.
Also saw an updated still maintained fork of MPC-HC called MPC-BE but it's windows only.

**Comment by wm4** [comment]
It's still dead. This project has nothing to do with MPC-HC, other than having similar name and GUI.

**Comment by maor26** [comment]
Yeah, I know this project has nothing to do with MPC-HC, was just looking for a good player with similar gui/feature set and was hoping to find an active project.
Guess i'll just keep looking.

**Comment by ahjolinna** [comment]
well this sucks, it seems mpc-qt fails to build already ...sigh
I assume problem is with latest mpv(-git) changes 

```
[  155s] g++ -c -pipe -O2 -Wall -D_FORTIFY_SOURCE=2 -fstack-protector-strong -funwind-tables -fasynchronous-unwind-tables -fstack-clash-protection -Werror=return-type -flto=auto -g -Wall -O2 -std=gnu++1y -Wall -Wextra -D_REENTRANT -fPIC -DMPCQT_PREFIX=\"/usr\" -DQT_NO_DEBUG -DQT_WIDGETS_LIB -DQT_X11EXTRAS_LIB -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_DBUS_LIB -DQT_CORE_LIB -I. -isystem /usr/include/qt5 -isystem /usr/include/qt5/QtWidgets -isystem /usr/include/qt5/QtX11Extras -isystem /usr/include/qt5/QtGui/5.14.1 -isystem /usr/include/qt5/QtGui/5.14.1/QtGui -isystem /usr/include/qt5/QtGui -isystem /usr/include/qt5/QtNetwork -isystem /usr/include/qt5/QtDBus -isystem /usr/include/qt5/QtCore/5.14.1 -isystem /usr/include/qt5/QtCore/5.14.1/QtCore -isystem /usr/include/qt5/QtCore -I. -I. -I/usr/lib64/qt5/mkspecs/linux-g++ -o ipcmpris.o ipcmpris.cpp
[  156s] In file included from ipcmpris.cpp:10:
[  156s] mpvwidget.h:13:10: fatal error: mpv/qthelper.hpp: No such file or directory
[  156s]    13 | #include <mpv/qthelper.hpp>
[  156s]       |          ^~~~~~~~~~~~~~~~~~
[  156s] compilation terminated.
```

**Comment by wm4** [comment]
https://github.com/wm4/mpc-qt/commit/c5da99e517887ee8870f84da59de98295d9670c9

**Comment by ahjolinna** [comment]
> [wm4/mpc-qt@c5da99e](https://github.com/wm4/mpc-qt/commit/c5da99e517887ee8870f84da59de98295d9670c9)

oh cool, love you (no homo) :D

**Comment by likrtr** [comment]
Hello. Is the GitLab mirror dead? Asking the question because I have opened the pull request there?
https://gitlab.com/mpc-qt/mpc-qt/-/merge_requests/1

**Comment by lgbaldoni** [comment]
@likrtr I maintain that gitlab repo, it's a snapshot to the latest original commit by CmdrKotori.
For some reason I didn't receive any notification for that, will investigate.

**Comment by lgbaldoni** [comment]
@wm4 @ahjolinna wrt the gitlab repo, what do you think ? Should it be a mirror of either of your forks?
Or vice versa and keep the development there? Or just leave it as it is with an additional note saying that anyone interesting in further improvements should come here?

**Comment by wm4** [comment]
Complete history is of course preferable. My fork has a commit that fixes mpc-qt build with the coming mpv release, but I won't maintain a Qt program and will chase anyone away who wants my repo to be the "official" one. (Feel free to cherry-pick the commit, though.)

**Comment by likrtr** [comment]
Hey. It seems that the original creator has come back and resumed the work on the project.
https://github.com/cmdrkotori/mpc-qt-origin

**Comment by lgbaldoni** [comment]
@likrtr [my thoughts now](https://youtu.be/ttEMYvpoR-k?t=38).

**Comment by cmdrkotori** [comment]
Hi guys. I'm moving development to https://github.com/mpc-qt/mpc-qt.
